### DISCUSSION TITLE
Disable / Enable "motion warning"

### DISCUSSION THREAD

**Discussion by JamieP83** [tsightler/ring-mqtt#569] [Feature Requests]
Hey all,
not sure if possible and i did search to no avail.
is it possible to somehow enable "Motion Warning" (not motion alerts) somehow?

its a really useful feature (at night) but not in the day!

**Comment by eugeneniemand** [comment]
Agreed, I think they are still valid in the day but I would do something like "If the door has been open for more than 2 minutes disable the warning until the door is closed again" So if I'm busy going in and out for an extended period of time I dont want to be told every 2 minutes that I'm being recorded

**Comment by tsightler** [comment]
For future reference, API to enable/disable motion warnings:

https://api.ring.com/clients_api/doorbots/<doorbot_id>/motion_announcement?motion_announcement=false
https://api.ring.com/clients_api/doorbots/<doorbot_id>/motion_announcement?motion_announcement=true

And in settings the current state can be determined by the following property:
`"motion_announcement": true`

The one thing that is not completely clear is how to determine which devices have this support vs those that don't.  It appears that it is supported on pretty much all line powered devices as all of my cameras support it and I don't own battery/low-power devices.

**Reply by JamieP83** [reply]
Hey, im happy to help if needed?
i have various ring cameras but it seems that its mostly the spotlights that have this feature built in

**Reply by tsightler** [reply]
Ring says that these cameras support this feature:

- Stick Up Cam Elite
- Stick Up Cam Plug In (3rd Gen)*
- Indoor Cam
- Spotlight Cam Wired
- Spotlight Cam Mount
- Floodlight Cam
- Floodlight Cam Wired Pro
- Floodlight Cam Wired Plus
- Ring Video Doorbell Pro
- Ring Video Doorbell Pro 2 
- Ring Video Doorbell Elite

But I don't really want to have to list all of these in the code.  I'm wondering if I can just query if the motion_announcement property exists in the device data.  Do you own any cameras that don't have this feature?  If so, running the data discovery tool could help:

https://github.com/dgreif/ring/wiki/Data-Discovery

**Comment by JamieP83** [comment]
Hey!
I have 14 different cameras (various types) and ALL of those have the “motion warning” im afraid and are listed below.
Could you maybe add the feature and then add a comment that it might not work on legacy devices or something?

**Comment by tsightler** [comment]
> Hey! I have 14 different cameras (various types) and ALL of those have the “motion warning” im afraid and are listed below. Could you maybe add the feature and then add a comment that it might not work on legacy devices or something?

Maybe, but any time I introduce a feature and it shows up for cameras that don't have that feature people complain about it or even open up issues reporting it, so I try not to do this.

Looking at that list, it doesn't really seem to be related to legacy devices as all of my devices are 5+ years old.  I think it's more low-power vs high-power devices as all of the devices above are high-power devices that use line power.  Maybe I can just use that.

**Comment by JamieP83** [comment]
Good idea!


Thanks,
Jamie

Sent from my iPhone

On 13 Jul 2023, at 9:54 pm, tsightler ***@***.***> wrote:

﻿

Hey! I have 14 different cameras (various types) and ALL of those have the “motion warning” im afraid and are listed below. Could you maybe add the feature and then add a comment that it might not work on legacy devices or something?

Maybe, but any time I introduce a feature and it shows up for cameras that don't have that feature people complain about it or even open up issues reporting it, so I try not to do this.

Looking at that list, it doesn't really seem to be related to legacy devices as all of my devices are 5+ years old. I think it's more low-power vs high-power devices as all of the devices above are high-power devices that use line power. Maybe I can just use that.

**Comment by eugeneniemand** [comment]
Is it possible for me to to call the API manually until this is implemented? Is it hard to do? Alternatively do you have a feature branch that I can use?

**Reply by tsightler** [reply]
I posted the API endpoints above so sure, you can call them however you wish, there is nothing hidden.  Like anything whether it is hard to do or not is dependent on your knowledge, if you understand REST APIs, then it is easy, if not, it is probably hard.

Right now there is no branch with this feature as there are other more critical issues that I'm currently reasearching in the limited time I have to work on this project, mainly the HEVC issue (I've made some progress here thanks to a user sharing their camera).  Overall, this feature would be easy to implement when I manage to get time for it, probably would take <30 minutes to code it up and test it.  It's certainly likely to be in the next version sometime in the coming weeks.

**Comment by tsightler** [comment]
v5.5.0 has been released with this feature included.

**Comment by JamieP83** [comment]
amazing! thank  you so much
